wait_for/__init__.py

```diff
@@ -104,36 +124,46 @@ def wait_for(func, func_args=[], func_kwargs={}, logger=None, **kwargs):
         correctly, only that it returned correctly at last check.
 
     Args:
-        func: A function to be run
-        func_args: A list of function arguments to be passed to func
-        func_kwargs: A dict of function keyword arguments to be passed to func
-        num_sec: An int describing the number of seconds to wait before timing out.
-        timeout: Either an int describing the number of seconds to wait before timing out. Or a
-            :py:class:`timedelta` object. Or a string formatted like ``1h 10m 5s``. This then sets
-            the ``num_sec`` variable.
-        expo: A boolean flag toggling exponential delay growth.
-        message: A string containing a description of func's operation. If None,
-            defaults to the function's name.
-        fail_condition: An object describing the failure condition that should be tested
-            against the output of func. If func() == fail_condition, wait_for continues
-            to wait. Can be a callable which takes the result and returns boolean whether to fail.
+        func (callable): A function to be run
+        func_args (Iterable[Any]): A list of function arguments to be passed to func
+        func_kwargs (dict[str, Any]): A dict of function keyword arguments to be passed to func
+        num_sec (int): An int describing the number of seconds to wait before timing out.
+        timeout (Union[int, timedelta, str]): Describes time to wait before timing out.
+            Either an int describing the number of seconds.
+            Or a :py:class:`timedelta` object.
+            Or a string formatted like ``1h 10m 5s``.
+            This then sets the ``num_sec`` variable.
+        expo (Any): A flag toggling exponential delay growth.
+        message (Optional[str]): A description of func's operation. If None, defaults to the
+            function's name.
+        fail_condition (Union[callable, Any, set[Any]]): An object describing the failure
+            condition that should be tested against the output of func.
+            If func() == fail_condition, wait_for continues to wait.
+            Can be a callable which takes the result and returns boolean whether to fail.
             You can also specify it as a  set, that way it checks whether it is present in the
             iterable.
-        handle_exception: A boolean controlling the handling of excepetions during func()
-            invocation. If set to True, in cases where func() results in an exception,
+        handle_exception(Union[Type[Exception], Iterable[Type[Exception]], Any]):
+            A parameter the handling of excepetions during func() invocation.
+            If set to ``Union[Type[Exception], Iterable[Type[Exception]]`` clobber exception
+            just from listed exceptions and treat it as a fail_condition.
+            If could be casted to True, in cases where func() results in an exception,
```

Comment:
Can I suggest some different phrasing here, to describe the boolean case for handle_exception.  Something like the following, adjust as you see fit.  The context of exceptions during func() is described above, so this can be shortened.

`If set to \`\`Boolean\`\` True, clobber any exception and treat it as a fail_condition. Defaults to False, which will not clobber any exceptions`

Reply:
https://docs.python.org/3/reference/expressions.html#booleans

In the context of Boolean operations, and also when expressions are used by control flow statements, the following values are interpreted as false: False, None, numeric zero of all types, and empty strings and containers (including strings, tuples, lists, dictionaries, sets and frozensets). **All other values are interpreted as true**. User-defined objects can customize their truth value by providing a __bool__() method.

Let leave as is to do this in an idiomatic way)